pkg/executables/kubectl.go

```diff
@@ -1869,6 +1869,46 @@ func (k *Kubectl) ValidateNodesVersion(ctx context.Context, kubeconfig string, k
 	return nil
 }
 
+// ValidateClusterNodeVersion validates the Control Plane node's kubernetes version in E2E tests.
+func (k *Kubectl) ValidateClusterNodeVersion(ctx context.Context, kubeconfig string, kubeVersion v1alpha1.KubernetesVersion) error {
```

Comment:
I don't think this belongs here. This kubectl should be just a client that reads and writes data to the api server. The validation IMO should be somewhere else.

Reply:
I would suggest just using `List` for example (which I think it's already implemented). And do the validation in your test code.

It might even be easier if you just use a client.Client instead.